MAVProxy/modules/lib/wxconsole_ui.py

```diff
@@ -93,52 +93,52 @@ def on_timer(self, event):
             self.timer.Stop()
             self.Destroy()
             return
-        while state.child_pipe_recv.poll():
-            try:
+        try:
```

Comment:
Why are you shifting the `try` here?

It increases the code covered by the try/except block - when the exception being caught should really only come from the line that's currently covered by the try/except block, shouldn't it?

Reply:
Similar to https://github.com/ArduPilot/MAVProxy/commit/4898e10555f22a0db5a3e36be7243b568ca6c732, the ``BrokenPipeException`` occurs when ``poll()`` is called. Not sure if there's a cleaner way to do this...